aql-engine/src/main/java/org/ehrbase/openehr/aqlengine/service/AqlQueryServiceImp.java

```diff
@@ -188,29 +188,30 @@ private static AqlQuery buildAqlQuery(AqlQueryRequest aqlQueryRequest) {
         AqlQuery aqlQuery = AqlQueryParser.parse(aqlQueryRequest.queryString());
 
         // apply limit and offset - where the definitions from the aql are the precedence
-        Long fetch = aqlQueryRequest.fetch();
-        if (fetch != null) {
-            if (aqlQuery.getLimit() != null) {
-                throw new InvalidApiParameterException(
-                        "Invalid AQL query: fetch is defined on query %s and as parameter %s"
-                                .formatted(aqlQuery.getLimit(), fetch));
-            }
-            aqlQuery.setLimit(fetch);
-        }
-        Long offset = aqlQueryRequest.offset();
-        if (offset != null) {
-            if (aqlQuery.getOffset() != null) {
-                throw new InvalidApiParameterException(
-                        "Invalid AQL query: fetch is defined on query %s and as parameter %s"
-                                .formatted(aqlQuery.getOffset(), offset));
-            }
-            aqlQuery.setOffset(offset);
+        Long fetchParam = aqlQueryRequest.fetch();
+        Long offsetParam = aqlQueryRequest.offset();
+        Long limitQuery = aqlQuery.getLimit();
+
+        // verify not parameter fetch offset are defined when query contains a LIMIT or assign fetch parameter
+        Optional.ofNullable(limitQuery)
+                .ifPresentOrElse(
+                        limit -> {
+                            if (fetchParam != null || offsetParam != null) {
+                                throw new UnprocessableEntityException(
+                                        "Query contains a LIMIT clause, fetch and offset parameters must not be used");
+                            }
+                        },
+                        () -> Optional.ofNullable(fetchParam).ifPresent(aqlQuery::setLimit));
+
+        // assign offset parameter
+        if (aqlQuery.getOffset() == null) {
+            Optional.ofNullable(offsetParam).ifPresent(aqlQuery::setOffset);
         }
```

Comment:
This feels overly complex, maybe try the following: 
if limitQuery null: assign from parameters
else: check parameters are null

Reply:
done